app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :user, class_name: 'User'
+
+  def update_posts_counter
+    user.increment!(:post_counter)
+  end
```

Comment:
- [ Optional ] Hi as a suggestion, you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private